### PR TITLE
feat: add vue-ts vue3-ts template

### PR DESCRIPTION
**Pull request by jerrywu001**
## What kind of change does this pull request introduce?

image

Feature:

- add vue-ts vue3-ts template
- update vue vue3 template

I checked it and it works fine.

@danilowoz  In daily development, the VUE3-TS is widely used. It is recommended to merge.

### CODE REVIEW

**Comment by jerrywu001** [comment]
In daily development, the VUE3-TS is widely used. It is recommended to merge.

**Comment by jerrywu001** [comment]
image

I don't have permission. 🤣🤣

**Comment by danilowoz** [comment]
Hey @jerrywu001, I loved your contribution! But let's keep the minimal necessary files to run a project, so I'd recommend removing the button and counter and leaving only a Hello world. Maybe the word `world` can come from a variable.

What do you think?

**Comment by jerrywu001** [comment]
> Hey @jerrywu001, I loved your contribution! But let's keep the minimal necessary files to run a project, so I'd recommend removing the button and counter and leaving only a Hello world. Maybe the word `world` can come from a variable.
> 
> What do you think?

Ok, let me do some change.

**Comment by jerrywu001** [comment]
> Hey @jerrywu001, I loved your contribution! But let's keep the minimal necessary files to run a project, so I'd recommend removing the button and counter and leaving only a Hello world. Maybe the word `world` can come from a variable.
> 
> What do you think?

Ok, let me do some change.

**Comment by jerrywu001** [comment]
> Hey @jerrywu001, I loved your contribution! But let's keep the minimal necessary files to run a project, so I'd recommend removing the button and counter and leaving only a Hello world. Maybe the word `world` can come from a variable.
> 
> What do you think?

@danilowoz  Done.

File: sandpack-react/src/templates/vue-ts.ts [hunk 1]
```diff
@@ -0,0 +1,123 @@
+/**
```
Comment (danilowoz) [inline sandpack-react/src/templates/vue-ts.ts:1]: I was wondering, is there any reason why we still maintain `vue` instead only `vue3`? At some point in the future we can sunset it and provide only the latest version of Vue

File: sandpack-react/src/templates/vue3.ts [hunk 2]
```diff
@@ -4,40 +4,20 @@
...
-      code: `<script setup>
-import { ref } from "vue";
-
-const msg = ref("world");
-</script>
-
-<template>
+      code: `<template>
   <h1>Hello {{ msg }}</h1>
 </template>
```
Comment (danilowoz) [inline sandpack-react/src/templates/vue3.ts:9]: Missing a line break between `template` and `script`?

File: sandpack-react/src/templates/vue-ts.ts (hunk 1 above)
Comment (danilowoz) [reply sandpack-react/src/templates/vue-ts.ts:1]: Yeah, we spoke internally, and let's focus on `vue3` only. This means we can remove this file and keep the original `vue` template, and on the next major release, we'll remove the `vue` template in favor of the `vue3`.

**Review by danilowoz — APPROVED** [review]
Thank you!

**Pull request merged**